src/Illuminate/Encryption/OpenSslEncrypter.php

```diff
@@ -55,19 +55,32 @@ public static function supported($key, $cipher)
     {
         $length = mb_strlen($key, '8bit');
 
-        return ($cipher === 'AES-128-CBC' && $length === 16) ||
-               ($cipher === 'AES-256-CBC' && $length === 32);
+        return isset(self::KEY_LENGTHS[$cipher]) && $length === self::KEY_LENGTHS[$cipher];
     }
 
     /**
      * Create a new encryption key for the given cipher.
      *
      * @param  string  $cipher
      * @return string
+     *
+     * @throws \Exception
      */
-    public static function generateKey($cipher)
+    public function generateKey($cipher = null)
     {
-        return random_bytes($cipher === 'AES-128-CBC' ? 16 : 32);
+        $cipher = $cipher ?? $this->cipher;
+
+        return random_bytes($cipher === self::AES_128 ? 16 : 32);
```

Comment:
We can use `self::KEY_LENGTHS[$cipher]`.

Reply:
Good call, I changed that and added a check whether the cipher is supported